rust/ql/lib/codeql/rust/frameworks/Poem.qll

```diff
@@ -5,16 +5,18 @@
 private import rust
 private import codeql.rust.Concepts
 private import codeql.rust.dataflow.DataFlow
+private import codeql.rust.internal.TypeInference
+private import codeql.rust.internal.Type
 
 /**
  * Parameters of a handler function
  */
 private class PoemHandlerParam extends RemoteSource::Range {
   PoemHandlerParam() {
-    exists(TupleStructPat param |
-      param.getResolvedPath() = ["crate::web::query::Query", "crate::web::path::Path"]
-    |
-      this.asPat().getPat() = param.getAField()
+    exists(TupleStructPat param, Type t |
+      this.asPat().getPat() = param.getAField() and
+      t = inferType(param) and
```

Comment:
Should now be removed (as well as the two imports).

Reply:
Sorry, the heat must be getting to me.  Now done.